storage/scripts/merkle_storage_stats.rs

```diff
@@ -0,0 +1,251 @@
+use std::convert::TryInto;
+use std::fs::File;
+use std::io::Read;
+
+use storage::*;
+// use context_action_storage::ContextAction;
+use backend::{
+    BTreeMapBackend, InMemoryBackend, KVStoreGCed, MarkSweepGCed, RocksDBBackend, SledBackend,
+};
+use merkle_storage::{check_commit_hashes, Entry, EntryHash, MerkleStorage};
+use storage_backend::StorageBackend;
+
+mod actions_tool;
+use actions_tool::{ActionsFileReader, ContextAction};
+
+fn parse_mem_value(value: &str) -> usize {
+    let mut pair = value.split_whitespace();
+    let value: usize = pair.next().unwrap().parse().unwrap();
+
+    match pair.next() {
+        Some("kB") => value * 1024,
+        _ => panic!("unknown memory usage unit."),
+    }
+}
+
+#[cfg(target_os = "linux")]
+fn mem_usage() -> usize {
+    let mut text_buf = Default::default();
+
+    File::open("/proc/self/status")
+        .and_then(|mut f| f.read_to_string(&mut text_buf))
+        .unwrap();
+
+    let mut res = 0;
+
+    for line in text_buf.lines() {
+        let mut pairs = line.split(':');
+        match (pairs.next(), pairs.next()) {
+            // (Some("VmPeak"), Some(text)) => virt_peak = parse_mem_value(text),
+            // (Some("VmSize"), Some(text)) => virt = parse_mem_value(text),
+            (Some("VmRSS"), Some(text)) => res = parse_mem_value(text),
+            // (Some("VmSwap"), Some(text))
+            _ => {}
+        }
+    }
+
+    res
+}
+
+#[cfg(not(target_os = "linux"))]
+fn mem_usage() -> usize {
+    0
+}
+
+struct Args {
+    backend: String,
+    test_integrity: bool,
+    preserved_cycles: usize,
+    cycle_block_count: u32,
+    actions_file: String,
+}
+
+impl Args {
+    pub fn read_args() -> Self {
+        let app = App::new("merkle_storage_stats")
+            .about("generate merkle storage statistics data")
+            .arg(Arg::with_name("preserved_cycles")
+                 .long("preserved-cycles")
+                 .help("last number of cycles which should be preserved")
+                 .default_value("7"))
+            .arg(Arg::with_name("cycle_block_count")
+                 .long("cycle-block-count")
+                 .help("amount of blocks in each cycle")
+                 .default_value("2048"))
+            .arg(Arg::with_name("test_integrity")
+                 .takes_value(false)
+                 .long("test-integrity")
+                 .help("validate integrity of all commits for last 7 cycles"))
+            .arg(Arg::with_name("actions_file")
+                 .required(true)
+                 .help("path to the actions.bin")
+                 .index(1))
+            .arg(Arg::with_name("backend")
+                 .short("b")
+                 .long("backend")
+                 .default_value("in-memory-gced")
+                 .help("backend to use for storing merkle storage. Possible values: in-memory-gced, in-memory-mark-sweep-gced")
+                       // + "rocksdb, sled, in-memory-gced, in-memory-mark-sweep-gced")
```

Comment:
What is this comment for?

Reply:
Right now that script only supports in-memory stores. It probably should support rocksdb and sled as well, but those implementations don't implement measurements for memory usage, so it won't create relevant statistics.

Probably we should just  replace it with todo comment.

Reply:
Yes, please. Add a meaningful comment or remove that one if there is nothing to say.